### PR TITLE
Add `plotlyServerUrl` to `baseUrl`

### PR DESCRIPTION
**Pull request by chriddyp**
Fixes https://github.com/plotly/plotly.js/issues/2759#issuecomment-400776193

I haven't tested this.

### CODE REVIEW

**Comment by etpinard** [comment]
Ok. I'll take it from here.

Will be part of `v1.39.0`.

File: test/jasmine/tests/config_test.js [hunk 1]
```diff
@@ -470,7 +470,33 @@ describe('config argument', function() {
...
+                expect(gd._context.plotlyServerUrl).toBe('https://plot.ly');
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        it('can be set to other base urls', function(done) {
+            Plotly.plot(gd, [], {}, {plotlyServerUrl: 'dummy'})
+            .then(function() {
+                expect(gd._context.plotlyServerUrl).toBe('dummy');
```
Comment (etpinard) [inline test/jasmine/tests/config_test.js]: These tests aren't great, but I couldn't find an easy way to mock the form submit in 

https://github.com/plotly/plotly.js/blob/bcab36e5c27c7227b7952b1ecc14becdd18104f0/src/plots/plots.js#L214-L245

File: test/jasmine/tests/config_test.js [hunk 2]
```diff
@@ -474,9 +474,13 @@ describe('config argument', function() {
 
     describe('plotlyServerUrl:', function() {
         var gd;
+        var form;
 
         beforeEach(function() {
             gd = createGraphDiv();
+            spyOn(HTMLFormElement.prototype, 'submit').and.callFake(function() {
+                form = this;
```
Comment (etpinard) [inline test/jasmine/tests/config_test.js]: thank you @alexcjohnson for the tip!

File: src/plot_api/plot_config.js [hunk 3]
```diff
@@ -21,9 +21,9 @@ module.exports = {
     // no interactivity, for export or image generation
     staticPlot: false,
 
-    // base url for the 'Edit in Chart Studio' (aka sendDataToCloud) mode bar button
+    // base URL for the 'Edit in Chart Studio' (aka sendDataToCloud) mode bar button
     // and the showLink/sendData on-graph link
-    plotlyServerUrl: 'https://plot.ly',
+    plotlyServerURL: 'https://plot.ly',
```
Comment (alexcjohnson) [inline src/plot_api/plot_config.js]: good call, that also seems more standard in the built-in API (eg `canvas.toDataURL`) though we have some other internal routines that use `...Url`, like `Drawing.setClipUrl`.

**Comment by alexcjohnson** [comment]
Big improvement - thanks @chriddyp, looks great @etpinard 💃

**Pull request merged**